datadog-lambda.gemspec

```diff
@@ -32,6 +32,14 @@ Gem::Specification.new do |spec|
 
   spec.add_dependency 'aws-xray-sdk', '~> 0.11.3'
   spec.add_dependency 'dogstatsd-ruby', '~> 5.0'
+
+  ruby_version = Gem::Version.new(RUBY_VERSION) # rubocop:disable Gemspec/RubyVersionGlobalsUsage
+  if ruby_version >= Gem::Version.new('3.3')
+    spec.add_dependency 'nokogiri', '~> 1.16.0'
+  else
+    spec.add_dependency 'nokogiri', '~> 1.15.6'
+  end
+
```

Comment:
Unfortunately nokogiri 1.15.6 does not work for ruby 3.3, so I had to explicitly set the nokogiri version when building the gem for ruby greather than or equal to 3.3.
Let me know if you find another way to handle this issue

Reply:
I think we can upgrade this, since I just deprecated Ruby 2.7 in another PR, let me test it

Reply:
Yeah, this is still needed

Reply:
there is an issue here, in that the Gem that was built and published onto Rubygems was built using 3.2, so the requirement in the Gemfile is set to `~> 1.15.6` (which has a CVE) - because this is in the gemspec even when installed on Ruby 3.3 it's set to this at compile time (see the dependencies list at

Reply:
good cath @cnorthwood! 

I think that is an issue due to the deployment step and not an issue with the conditional setting of the `nokogiri` gem based in the `RUBY_VERSION` variable.

Executing locally, if the `gem build` runs having `ruby v 3.3.0` it successfully installed the correct nokogiri version 

![image](https://github.com/user-attachments/assets/fc506f8d-920d-402b-a550-d8f0987b6978)

I am not sure how @duncanista deploys this but I am guessing is using the script `publish_prod.sh`

![image](https://github.com/user-attachments/assets/f3237e35-380e-44a6-b036-272e08b20c4e)

However, I think the published layer is ok, for ruby 3.3 is using the correct nokogiri version, this is because the layer is built into a docker container where RUBY_VERSION is set properly based on the docker image

![image](https://github.com/user-attachments/assets/12672ae3-4667-4042-8130-8f7a47967528)

I have the new layer working properly in a production project, this was not possible before this release.

Reply:
I'll take a look tomorrow – thanks for bringing this out, I'll also make sure to have a hotfix on the Gem, I think indeed the layers are OK whereas the gem is probably using my local ruby, which I think I might have messed up – sorry.

Reply:
Updating this in #97

Reply:
Fixed in [v2.23.1](https://github.com/DataDog/datadog-lambda-rb/releases/tag/v2.23.1)